src/Symfony/Component/Validator/Mapping/Loader/AbstractLoader.php

```diff
@@ -78,6 +78,14 @@ protected function newConstraint(string $name, mixed $options = null): Constrain
             $className = self::DEFAULT_NAMESPACE.$name;
         }
 
-        return new $className($options);
+        try {
+            return new $className($options);
+        } catch (\TypeError $e) {
+            if (!\is_array($options) || \PHP_VERSION_ID < 80100) {
+                throw $e;
+            }
+
+            return new $className(...$options);
+        }
```

Comment:
I really don't think we should use a `TypeError` for normal application flow. That error could indicate a real problem, that we would hide this way. There has to be a better way.

Reply:
I can use reflection then to check if there is required args if that works for you